File: csharp/console/euler/src/euler/003/Problem003.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace euler._003
{
    /// <summary>
    /// The prime factors of 13195 are 5, 7, 13 and 29. What is the largest prime factor of the number 600851475143 ?
    /// </summary>
    public class Problem003 : ISolution
    {
        public string Solution()
        {
            var number = 600851475143;
            var list = new List<long>();

            while (number % 2 == 0)
            {
                list.Add(number);
                number = number / 2;
            }

            for (int i = 3; i <= Math.Sqrt(number); i = i + 2)
            {
                while (number % i == 0)
                {
                    list.Add(number);
                    number = number / i;
                }
            }

            if (number > 2)
            {
                list.Add(number);
            }

            // Linq extensions
            return string.Format("{0}", list.Min());
        }
    }
}

File: golang/src/problem-003.go
// Project Euler - Problem 003
// Joao Antao

// The prime factors of 13195 are 5, 7, 13 and 29.
// What is the largest prime factor of the number 600851475143 ?

package main
import ("fmt"; "runtime")

func Solution(n int) int {
  factor := 2

  for n > factor {
    if n % factor == 0 {
      n /= factor
      factor = 2
    } else {
      factor += 1
    }
  }
  return n
}

func main() {
    number := 600851475143
    os := runtime.GOOS

    defer fmt.Println(Solution(number))
    fmt.Println("Operating system: ", os)
}
